### ISSUE TITLE
Using teleport with shallowMount

### ISSUE THREAD

**Issue by doutatsu** [vuejs/vue-test-utils-next#1070]
I am getting a Vue warning for my teleport, when doing a shallowMount. I am unable to find a way to get a workaround on this - tried stubbing it out, but that also didn't do anything...

### Warning
```js
console.warn
    [Vue warn]: Non-function value encountered for default slot. Prefer function slots for better performance.
      at <AnonymousStub to="nav" >
      at <Anonymous ref="VTU_COMPONENT" >
      at <VTUROOT>
```

### Component template
```pug
<template lang="pug">
teleport(to='nav')
  .fixed.bottom-5.right-5(ref='fab')
    span.relative.z-50
      button.fab.ripple(
        @click='open = !open'
        type='button'
        aria-haspopup='true'
        aria-expanded='false'
      )
        .absolute.transform.transition.duration-150(
          :class="open ? 'rotate-90 opacity-0' : 'rotate-0 opacity-100'"
        )
          slot
        icon-x-outline.absolute.w-6.h-6.inline-block.text-white.transform.transition.duration-150(
          ref='toggleIcon'
          style="padding: 2px 0 0 2px"
          :class="open ? 'rotate-0 opacity-100' : '-rotate-90 opacity-0'"
        )
    transition(name="overlay-fade")
      button.click-off-button(
        ref="clickOffButton"
        v-show='open'
        @click='open = false'
        tabindex='-1'
      )
        .absolute.inset-0.bg-white.opacity-95.dark_bg-gray-900.dark_opacity-90
    template(v-if="options.length")
      transition(name="dropdown")
        .dropdown-body.w-full.sm_w-40(v-show='open')
          .py-1
            a.group.justify-end.space-x-4(
              v-for="option in options"
              href='#'
              @click.prevent="handleClick(option.action)"
            )
              span(v-if="option.text" v-text="option.text")
              .secondary-fab.ripple
                component.h-5.w-5.text-gray-400.dark_text-white(:is="option.icon")
</template>
```

### Test
```js
import BaseFab from '@/components/base_components/BaseFab.vue';

describe('BaseFab.vue', () => {
  let baseFab;

  describe('when open', () => {
    beforeEach(() => {
      baseFab = shallowMount(BaseFab, {
        slots: { default: '<div>My slot content</div>' },
        data() { return { open: true }; },
      });
    });

    it('shows cross icon', () => {
      expect(baseFab.findComponent({ ref: 'toggleIcon' }).element)
        .toHaveClass('opacity-100');
    });
  });
});
```

**Comment by lmiller1990** [comment]
Testing teleport is a bit weird. There is an article you might find useful here: https://next.vue-test-utils.vuejs.org/guide/advanced/teleport.html

As for the warning, I am not sure that syntax is valid. I don't think we support passing a template like that. [Docs for slots mounting option](https://next.vue-test-utils.vuejs.org/api/#slots). I think you want something like:

```js
baseFab = shallowMount(BaseFab, {
  slots: { default: h('div', 'My slot content</div>' },
  data() { return { open: true } }
});
```

I don't think there is an actual bug in Test Utils, so I'm going to close this for now to ensure issues don't get out of control. Happy to help debug and get your test working, and re-open if there's actually a bug.

If you could make your test a bit more minimal (ideally provide a repo I can run) that would help even more. Can you try the above syntax and see what happens? I think you will still have a problem: there's no `nav` to teleport to in your test, but you can make one ([see the docs](https://next.vue-test-utils.vuejs.org/guide/advanced/teleport.html) or a video version of the same content [here](https://www.youtube.com/watch?v=bhYtFWDgJmY&ab_channel=LachlanMiller).

**Comment by doutatsu** [comment]
@lmiller1990 Thanks for a quick response - I've already read the guide for teleport, and none of the suggestions really worked for me. 

I can also confirm that the syntax I used did work - I can see it when using `html()` on the component. But I did actually try `h` version before, as I was researching this error and someone suggested it needs to be an actual VNode... but that didn't help either.

Due to the shallow mount, I don't think it matters whatever `nav` exists or not - I think Vue doesn't like the whole template being passed to `teleport` as it must be using a slot underneath the teleport component. I'll give a stab at making a reproducible repo
